### PR TITLE
Crowdsale ico

### PR DESCRIPTION
**Pull request by DBarinovv**
- Add tests
- Remove deprecated function
- Change `unwrap()` to `expect()`

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,674 +1,674 @@
...
-For more information on this, and how to apply and follow the GNU GPL, see
-<https://www.gnu.org/licenses/>.
-
-  The GNU General Public License does not permit incorporating your program
-into proprietary programs.  If your program is a subroutine library, you
-may consider it more useful to permit linking proprietary applications with
-the library.  If this is what you want to do, use the GNU Lesser General
-Public License instead of this License.  But first, please read
-<https://www.gnu.org/licenses/why-not-lgpl.html>.
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (shamilsan) [inline LICENSE]: Need replace with the MIT license.

File: Cargo.toml [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-gtest = { git = "https://github.com/gear-tech/gear.git" }
-
-[build-dependencies]
-gear-wasm-builder = { git = "https://github.com/gear-tech/gear.git" }
-
+[package]
+name = "crowdsale_ico"
+version = "0.1.0"
+edition = "2021"
+license = "GPL-3.0"
```
Comment (shamilsan) [inline Cargo.toml]:
```suggestion
license = "MIT"
```

File: tests/balance_tests.rs [hunk 3]
```diff
@@ -22,7 +21,11 @@ fn balance_after_two_purchases() {
 
     balance_of(&ico, amount);
 
-    sys.spend_blocks((TIME_INCREASE_STEP + 1).try_into().unwrap());
+    sys.spend_blocks(
+        (TIME_INCREASE_STEP + 1)
+            .try_into()
+            .expect("Can't cast type"),
+    );
```
Comment (shamilsan) [inline tests/balance_tests.rs]: Use `as`.
```suggestion
    sys.spend_blocks((TIME_INCREASE_STEP + 1) as _);
```

File: tests/buy_tokens_tests.rs [hunk 4]
```diff
@@ -55,12 +55,16 @@ fn buy_tokens_after_price_update() {
 
     start_sale(&ico, 2);
 
-    sys.spend_blocks((TIME_INCREASE_STEP).try_into().unwrap());
+    sys.spend_blocks((TIME_INCREASE_STEP).try_into().expect("Can't cast type"));
```
Comment (shamilsan) [inline tests/buy_tokens_tests.rs]:
```suggestion
    sys.spend_blocks(TIME_INCREASE_STEP as _);
```

File: tests/buy_tokens_tests.rs [hunk 5]
```diff
@@ -55,12 +55,16 @@ fn buy_tokens_after_price_update() {
...
 
     let amount: u128 = 5;
     buy_tokens(&ico, amount, amount * (START_PRICE + PRICE_INCREASE_STEP));
 
-    sys.spend_blocks((TIME_INCREASE_STEP - 1).try_into().unwrap());
+    sys.spend_blocks(
+        (TIME_INCREASE_STEP - 1)
+            .try_into()
+            .expect("Can't cast type"),
+    );
```
Comment (shamilsan) [inline tests/buy_tokens_tests.rs]:
```suggestion
    sys.spend_blocks((TIME_INCREASE_STEP - 1) as _);
```

File: tests/init_ico.rs [hunk 6]
```diff
@@ -60,6 +61,7 @@ pub fn init(sys: &System) {
 
     init_fungible_token(sys);
     init_ico(sys);
+    sys.mint_to(USER_ID, 100000);
```
Comment (shamilsan) [inline tests/init_ico.rs]: To be more readable:
```suggestion
    sys.mint_to(USER_ID, 100_000);
```

**Review by shamilsan — APPROVED** [review]
Please pay attention to end-of-lines. You are to configure git to have CRLF only on your local machine, not on the git server. I've reverted EOLs back.

**Comment by shamilsan** [comment]
We need to merge the master branch after #9. I've messed up tests with the `mint_to` function, your solution is better, so one needs to revert it back 😞

**Review by LouiseMedova — APPROVED** [review]

**Pull request merged**